code/game/mecha/mecha.dm

```diff
@@ -945,6 +949,20 @@
 		else
 			step(obstacle,src.dir)
 
+	else if(istype(obstacle, /turf))
+		if(phasing && get_charge()>=phasing_energy_drain)
+			spawn()

```

Comment:
Should be splitting this out into a `set waitfor = FALSE` proc or using a timer/INVOKE_ASYNC rather than spawning.